scene/main/node.cpp

```diff
@@ -1715,8 +1715,41 @@ void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
 		return;
 	}
 
+	bool preserve_owner = data.owner && (data.owner == p_parent || data.owner->is_ancestor_of(p_parent));
+	Node *owner_temp = data.owner;
+	List<const Node *> common_parents;
+
+	// If the new parent is related to the owner, find all children of the reparented node who have the same owner so that we can reassign them.
+	if (preserve_owner) {
+		List<const Node *> to_visit;
```

Comment:
...and maybe here, too (`LocalVector`).

Reply:
`LocalVector` is perfectly suited for a stack

Reply:
ok! removed the `const`ness and changed the `List`s to `LocalVector`s